test/e2e/specs/editor/plugins/inner-blocks-render-appender.spec.js

```diff
@@ -0,0 +1,129 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'RenderAppender prop of InnerBlocks', () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await requestUtils.activatePlugin(
+			'gutenberg-test-innerblocks-render-appender'
+		);
+	} );
+
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test.afterAll( async ( { requestUtils } ) => {
+		await requestUtils.deactivatePlugin(
+			'gutenberg-test-innerblocks-render-appender'
+		);
+	} );
+
+	test( 'Users can customize the appender and can still insert blocks using exposed components', async ( {
+		editor,
+		page,
+	} ) => {
+		await editor.insertBlock( {
+			name: 'test/inner-blocks-render-appender',
+		} );
+
+		const customAppender = page.locator( '.my-custom-awesome-appender' );
+
+		// Verify if the custom block appender text is the expected one.
+		await expect(
+			customAppender.getByText( 'My custom awesome appender' )
+		).toBeVisible();
```

Comment:
Can we use the `toHaveText`/`toContainText` assertion here?

```suggestion
		await expect( customAppender ).toHaveText( 'My custom awesome appender' );
```